src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -1697,6 +1697,56 @@ static QVariant fcnRasterValue( const QVariantList &values, const QgsExpressionC
   return std::isnan( value ) ? QVariant() : value;
 }
 
+static QVariant fcnRasterAttributes( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsRasterLayer *layer = QgsExpressionUtils::getRasterLayer( values.at( 0 ), parent );
+  if ( !layer || !layer->dataProvider() )
+  {
+    parent->setEvalErrorString( QObject::tr( "Function `raster_attributes` requires a valid raster layer." ) );
+    return QVariant();
+  }
+
+  int bandNb = QgsExpressionUtils::getNativeIntValue( values.at( 1 ), parent );
+  if ( bandNb < 1 || bandNb > layer->bandCount() )
+  {
+    parent->setEvalErrorString( QObject::tr( "Function `raster_attributes` requires a valid raster band number." ) );
+    return QVariant();
+  }
+
+  const double value = QgsExpressionUtils::getDoubleValue( values.at( 2 ), parent );
+  if ( std::isnan( value ) )
+  {
+    parent->setEvalErrorString( QObject::tr( "Function `raster_attributes` requires a valid raster value." ) );
+    return QVariant();
+  }
+
+  if ( ! layer->dataProvider()->attributeTable( bandNb ) )
+  {
+    return QVariant();
+  }
+
+  const QVariantList data = layer->dataProvider()->attributeTable( bandNb )->row( value );
+  if ( data.isEmpty() )
+  {
+    return QVariant();
+  }
+
+  QVariantMap result;
+  const QList<QgsRasterAttributeTable::Field> fields { layer->dataProvider()->attributeTable( bandNb )->fields() };
+  for ( int idx = 0; idx < static_cast<int>( fields.count( ) ) && idx < static_cast<int>( data.count() ); ++idx )
+  {
+    const QgsRasterAttributeTable::Field field { fields.at( idx ) };
+    if ( field.isColor() || field.isRamp() )
```

Comment:
Why do we remove these fields ?

Reply:
Because those fields (that can be from 0 up to 8) contain the color definition ( RGB(A) possibly min and max) of the style applied to the raster and you don't normally want to see them, they just add no information. The are also hidden from the identify results, but you can see and edit them in the raster attribute table.

Reply:
ok :+1: